cmd/gen-doc/gen-doc.go

```diff
@@ -57,8 +61,11 @@ var gadgetTemplate string
 
 func getTraceFactories() (ret []GadgetData) {
 	for name, factory := range gadgetcollection.TraceFactories() {
+		t := reflect.TypeOf(factory).Elem()
+		pkgPath := strings.TrimPrefix(t.PkgPath(), "github.com/kinvolk/inspektor-gadget/pkg/gadget-collection/gadgets/")
 		ret = append(ret, GadgetData{
 			Name:        name,
+			Pkg:         pkgPath,
```

Comment:
I think a better implementation would be to add a method `GetEventPrototype()` to the Factory and then use `reflect` to find the package name. That would avoid to hard code the string replacement of `pkg/gadget-collection/gadgets/` by `pkg/gadgets/.../types`.